crates/pallet-transaction-fees/src/lib.rs

```diff
@@ -0,0 +1,294 @@
+        let collected_storage_fees_reward = collected_fees.storage
+            / T::StorageFeesEscrowBlockTax::get().1.into()
+            * (T::StorageFeesEscrowBlockTax::get().1 - T::StorageFeesEscrowBlockTax::get().0)
+                .into();
+        storage_fees_escrow += collected_fees.storage - collected_storage_fees_reward;
+
+        // Update storage fees escrow.
+        if storage_fees_escrow != original_storage_fees_escrow {
+            CollectedStorageFeesEscrow::<T>::put(storage_fees_escrow);
+            Self::deposit_event(Event::<T>::StorageFeesEscrowChange {
+                before: original_storage_fees_escrow,
+                after: storage_fees_escrow,
+            });
+        }
+
+        // Issue storage fees reward.
+        let storage_fees_reward = storage_fees_escrow_reward + collected_storage_fees_reward;
+        if !storage_fees_reward.is_zero() {
+            T::Currency::deposit_into_existing(&block_author, storage_fees_reward).expect(
+                "Farmer account must have already received the block reward before fees are \
+                collected; qed",
+            );
+            Self::deposit_event(Event::<T>::StorageFeesReward {
+                who: block_author.clone(),
+                amount: storage_fees_reward,
+            });
+        }
+
+        // Issue compute fees reward.
+        if !collected_fees.compute.is_zero() {
+            T::Currency::deposit_into_existing(&block_author, collected_fees.compute)
+                .expect("Executor account must already exist before they execute the block; qed");
+            Self::deposit_event(Event::<T>::ComputeFeesReward {
+                who: block_author.clone(),
+                amount: collected_fees.compute,
+            });
+        }
+
+        // Issue tips reward.
+        if !collected_fees.tips.is_zero() {
+            T::Currency::deposit_into_existing(&block_author, collected_fees.tips)
+                .expect("Executor account must already exist before they execute the block; qed");
+            Self::deposit_event(Event::<T>::TipsReward {
+                who: block_author,
+                amount: collected_fees.tips,
+            });
+        }
+    }
+
+    pub fn transaction_byte_fee() -> BalanceOf<T> {
+        if let Some(transaction_byte_fee) = TransactionByteFee::<T>::get() {
+            return transaction_byte_fee;
+        }
+
+        let credit_supply = T::CreditSupply::get();
+
+        let transaction_byte_fee = match T::TotalSpacePledged::get().checked_sub(
+            T::BlockchainHistorySize::get() * u64::from(T::MinReplicationFactor::get()),
+        ) {
+            Some(free_space) if free_space > 0 => credit_supply / BalanceOf::<T>::from(free_space),
+            _ => credit_supply,
+        };
+
+        // Cache value for this block.
+        TransactionByteFee::<T>::put(transaction_byte_fee);
+
+        transaction_byte_fee
+    }
+
+    pub fn distribute_transaction_fees(
```

Comment:
Could be more appropriate for me to use `note_transaction_fees` or something like that as there is no real _distribution_(transferring/minting) here.

Reply:
Done